### DISCUSSION TITLE
issue: how to trigger error with react-select defaultValue

### DISCUSSION THREAD

**Discussion by sunyeongchoi** [react-hook-form/react-hook-form#8164] [Q&A]
I want to trigger `react-hook-form` **error** when it is reset to Controlled Select defaultValue.
I use `react-select` library.
How can I trigger a `required` error while using select defaultValue together?

Any help would be appreciated. :)

codesandbox : https://codesandbox.io/s/react-hook-form-controller-react-select-forked-ukfp9d?file=/src/App.js

**Comment by FilipBrannlund** [comment]
Hey @sunyeongchoi 
Try this
``` 
 const handleReset = async () => {
    // setValue("reset", { value: "", label: "" });
    resetField("reset");
    await trigger("reset");
  };
```

https://codesandbox.io/s/react-hook-form-controller-react-select-forked-0cdqxk?file=/src/App.js:255-400

`resetfield` https://react-hook-form.com/api/useform/resetfield
`trigger` https://react-hook-form.com/api/useform/trigger

**Reply by Moshyfawn** [reply]
While it's a valid suggestion, calling [the `trigger` form method](https://react-hook-form.com/api/useform/trigger) won't set the error, as the select's `defaultValue` is
```js
defaultValue={{ value: '', label: '' }}
```

The `rules` like `required` are aligned with the native HTML Web validation, which checks for an empty `string` / `null` / `undefined`
